homeassistant/components/opower/coordinator.py

```diff
@@ -86,40 +86,63 @@ async def _async_update_data(
         # Because Opower provides historical usage/cost with a delay of a couple of days
         # we need to insert data into statistics.
         await self._insert_statistics()
-        return {forecast.account.utility_account_id: forecast for forecast in forecasts}
+        return {forecast.account.uuid: forecast for forecast in forecasts}
 
     async def _insert_statistics(self) -> None:
         """Insert Opower statistics."""
-        for account in await self.api.async_get_accounts():
+        accounts = await self.api.async_get_accounts()
+
+        # Utility account id is not necessarily unique if there are multiple
+        # meters for a single account.  If there are 2 meters that have the
+        # same utility account id, then it is a multi-meter account.
+        utility_account_ids = [account.utility_account_id for account in accounts]
+        multi_meter_utility_account_ids = [
+            i for i in set(utility_account_ids) if utility_account_ids.count(i) > 1
+        ]
+
+        for account in accounts:
+            is_multi_meter = (
+                account.utility_account_id in multi_meter_utility_account_ids
+            )
+
+            # For backwards compatibility, use the utility_account_id
+            # if there is only a single meter for the account.
```

Comment:
Instead of the backwards compatibility, could we modify existing devices and entities with the new ids?